### PR TITLE
[fix] mtu value was always set to default value

### PR DESCRIPTION
**Pull request by LarryCotarbais**
This patch fixes incorrect mtu value set in /etc/origin/node/node-config.yaml file.
Since system_facts refactoring, mtu value is always set to default value (1450) as val['ipv4'] doest not exist anymore.

### CODE REVIEW

File: roles/openshift_facts/library/openshift_facts.py [hunk 1]
```diff
@@ -463,13 +463,9 @@ def set_sdn_facts_if_unset(facts, system_facts):
...
 
-        for val in itervalues(system_facts):
-            if isinstance(val, dict) and 'mtu' in val:
-                mtu = val['mtu']
-
-                if 'ipv4' in val and val['ipv4'].get('address') == node_ip:
-                    facts['node']['sdn_mtu'] = str(mtu - 50)
-
+        if 'ansible_default_ipv4' in system_facts:
+            if system_facts['ansible_default_ipv4'].get('address') == node_ip and 'mtu' in system_facts['ansible_default_ipv4']:
```
Comment (vrutkovs) [inline roles/openshift_facts/library/openshift_facts.py]: I don't see the difference with current code: all `system_facts` are iterated, eventually `val` would be `ansible_default_ipv4` and the same code would be executed, isn't it?

Comment (vrutkovs) [reply roles/openshift_facts/library/openshift_facts.py]: A cleaner fix would be:
```diff
- if 'ipv4' in val and val['ipv4'].get('address') == node_ip:
+ if val.get('address') == node_ip: 
```

Comment (LarryCotarbais) [reply roles/openshift_facts/library/openshift_facts.py]: Preivously, system_facts was computed like this in openshift_facts.py __init__  :
           # pylint: disable=too-many-function-args,invalid-name
            self.system_facts = ansible_facts(module, ['hardware', 'network', 'virtual', 'facter'])  # noqa: F405
            additional_facts = {}
            for (k, v) in self.system_facts.items():
                additional_facts["ansible_%s" % k.replace('-', '_')] = v
            self.system_facts.update(additional_facts)

Now system_facts comes from roles/openshift_facts/vars/main.yml from vars_openshift_facts_system_facts.
vars_openshift_facts_system_facts:
  ansible_default_ipv4: "{{ vars.ansible_default_ipv4 }}"
  ansible_nodename: "{{ vars.ansible_nodename }}"
  ansible_fqdn: "{{ vars.ansible_fqdn }}"
  ansible_product_name: "{{ vars.ansible_product_name }}"
  ansible_product_version: "{{ vars.ansible_product_version }}"
  ansible_virtualization_type: "{{ vars.ansible_virtualization_type }}"
  ansible_virtualization_role: "{{ vars.ansible_virtualization_role }}"
  ansible_system_vendor: "{{ vars.ansible_system_vendor }}"

Comment (LarryCotarbais) [reply roles/openshift_facts/library/openshift_facts.py]: Would the following be cleaner ?:

```diff
-            if isinstance(val, dict) and 'mtu' in val:
-                mtu = val['mtu']
 
-                if 'ipv4' in val and val['ipv4'].get('address') == node_ip:
-                    facts['node']['sdn_mtu'] = str(mtu - 50)
+            if isinstance(val, dict):
+                if val.get('address') == node_ip:
+                    facts['node']['sdn_mtu'] = str(val.get('mtu') - 50)
```

Comment (vrutkovs) [reply roles/openshift_facts/library/openshift_facts.py]: Yep, looks cleaner :+1:

File: roles/openshift_facts/library/openshift_facts.py [hunk 2]
```diff
@@ -464,12 +464,11 @@ def set_sdn_facts_if_unset(facts, system_facts):
...
-            if isinstance(val, dict) and 'mtu' in val:
-                mtu = val['mtu']
-
-                if 'ipv4' in val and val['ipv4'].get('address') == node_ip:
-                    facts['node']['sdn_mtu'] = str(mtu - 50)
 
+            if isinstance(val, dict):
+                if val.get('address') == node_ip:
+                    facts['node']['sdn_mtu'] = str(val.get('mtu') - 50)
+
```
Comment (vrutkovs) [inline roles/openshift_facts/library/openshift_facts.py]:
```suggestion
```
```
./roles/openshift_facts/library/openshift_facts.py:471:1: W293 blank line contains whitespace
```

**Comment by vrutkovs** [comment]
Looks good, please squash the commits into one and we'll get it merged soon

**Comment by jcpowermac** [comment]
@LarryCotarbais can you fix the trailing whitespace please.

**Comment by vrutkovs** [comment]
Sorry, I attempted to force-push the fix (since it appears to be a serious problem on new installs, especially GCP), but Github has remove removed this commit.

Resubmitted this as #11707 11707

**Pull request closed without merging**